### ISSUE TITLE
Add support for Nvidia StateViewer

### ISSUE THREAD

**Issue by oscarbg** [apitrace/apitrace#331]
Hi,
seeing GTC 2015 presentation about Stateviewer (https://github.com/rchoetzlein/stateviewer) see it needs a modified apitrace to work.
It's here:
https://github.com/rchoetzlein/apitrace/tree/nvidia-apitrace
seems adds a few options to glretrace to dump raw state to inspect then with Stateviewer..
As changes seems very minimal and useful are you interested on pushing to master branch?..
thanks..

**Comment by jrfonseca** [comment]
Actually, @rchoetzlein contacted me privately shortly after those changes to get them merged.

I looked at the code then, but my condition to merge was that the analysis is static (doesn't need live access to OpenGL implementation) therefore could and should be refactored into a new standalone `cli/cli_stateview.cpp` module instead of piggy-backing on `retrace/*`, ie.,

    apitrace stateviewer foo.trace

Potentially this could be even written as a python script. Runtime would be slower but it would allow to code more complex analysis more easily.  In apitrace/scripts/unpickle.py there is a sample class "Counter" which counts the frequencies of call numbers. It would be easy to adapt this to do the same.  Anyway, Python vs C++ is a minor issue.

The fact is that retrace is too important for me, and I can't afford to let that module it take on additional responsibilities unnecessarily. As it could hamper development of other things that must be there.

In short, I never opposed integrating, but I don't have the time to do the requested cleanup myself.  So, unless there is somebody voluteering to take this on, no point keeping this open.

**Comment by rchoetzlein** [comment]
Hi,
After talking with jrfonseca, my goal was to create a separate command and linking project for the sln, for proper integration into master branch. That's still in the back of my mind, but I got pulled away to other things, and just had time to put together the GTC results.. Hoping to get back to this sometime.

For now, you can clone the nvidia-apitrace branch and use that as is. If you have any questions, you can ping me on that branch, or on the stateviewer (https://github.com/rchoetzlein).
